xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDSP.h

```diff
@@ -0,0 +1,1288 @@
+    {
+      return instance->toAddon.addonInstance->InputResampleProcess(handle, array_in, array_out, samples);
+    }
+
+    static inline int ADDON_InputResampleSampleRate(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle)
+    {
+      return instance->toAddon.addonInstance->InputResampleSampleRate(handle);
+    }
+
+    static inline float ADDON_InputResampleGetDelay(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle)
+    {
+      return instance->toAddon.addonInstance->InputResampleGetDelay(handle);
+    }
+
+    static inline unsigned int ADDON_PreProcessNeededSamplesize(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle, unsigned int mode_id)
+    {
+      return instance->toAddon.addonInstance->PreProcessNeededSamplesize(handle, mode_id);
+    }
+
+    static inline float ADDON_PreProcessGetDelay(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle, unsigned int mode_id)
+    {
+      return instance->toAddon.addonInstance->PreProcessGetDelay(handle, mode_id);
+    }
+
+    static inline unsigned int ADDON_PreProcess(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle, unsigned int mode_id, float **array_in, float **array_out, unsigned int samples)
+    {
+      return instance->toAddon.addonInstance->PreProcess(handle, mode_id, array_in, array_out, samples);
+    }
+
+    static inline AE_DSP_ERROR ADDON_MasterProcessSetMode(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle, AE_DSP_STREAMTYPE type, unsigned int mode_id, int unique_db_mode_id)
+    {
+      return instance->toAddon.addonInstance->MasterProcessSetMode(handle, type, mode_id, unique_db_mode_id);
+    }
+
+    static inline unsigned int ADDON_MasterProcessNeededSamplesize(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle)
+    {
+      return instance->toAddon.addonInstance->MasterProcessNeededSamplesize(handle);
+    }
+
+    static inline float ADDON_MasterProcessGetDelay(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle)
+    {
+      return instance->toAddon.addonInstance->MasterProcessGetDelay(handle);
+    }
+
+    static inline int ADDON_MasterProcessGetOutChannels(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle, unsigned long &out_channel_present_flags)
+    {
+      return instance->toAddon.addonInstance->MasterProcessGetOutChannels(handle, out_channel_present_flags);
+    }
+
+    static inline unsigned int ADDON_MasterProcess(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle, float **array_in, float **array_out, unsigned int samples)
+    {
+      return instance->toAddon.addonInstance->MasterProcess(handle, array_in, array_out, samples);
+    }
+
+    static inline const char* ADDON_MasterProcessGetStreamInfoString(AddonInstance_AudioDSP* instance, const ADDON_HANDLE handle)
+    {
+      instance->toAddon.addonInstance->m_streamInfoString = instance->toAddon.addonInstance->MasterProcessGetStreamInfoString(handle);
```

Comment:
Why the assignment? Does it make sense to return instance->toAddon.addonInstance->GetDSPVersion() imidiatly?